### DISCUSSION TITLE
FYI: Reolink 2020 firmware updates do not impact Neolink

### DISCUSSION THREAD

**Discussion by twistedddx** [thirtythreeforty/neolink#114] [General]
Numerous firmware updates for RLC-410 cameras include the change:
"Optimized network transmission protocol to improve network security "

Eg both of these:
12/11/2020 Firmware for Reolink IP Cameras (IPC_51516M) 
10/31/2020 Firmware for Reolink IP Cameras (IPC_51516M)

I can confirm neither of these updates made any noticeable change to the BC protocol on the local network.

I have not tested E1 cameras but they include:
"Optimized the network transmission protocol."
This may be the same change and may also not impact Neolink, but is not tested.

While on the subject of official Reolink firmware it appears the hi3535 & hi3536 NVR devices are no longer receiving regular updates but have received what may be a one off update to replace the flash web player for a newer html5 player.

RLN4-410 	20120700 	H3MB17
RLN8-410 	20120700 	H3MB02
RLN8-410-E 	20120701 	H3MB16
RLN16-410 	20120700 	H3MB02

Changes.log extract:
Trunk Version 1.0.255.1, until 2020-11-20
..
Trunk Version 1.0.255, until 2019-01-08 (sic) (2020-01-08)
..
Trunk Version 1.0.254, until 2019-12-31
..

While the NT98312 devices appear to be receiving more updates
RLN8-410 	20102144 	H3MB18 or N2MB02
RLN16-410 	20102145 	H3MB18

Changes.log extract:
Trunk Version 1.0.263, until 2020-09-30
..
Trunk Version 1.0.262, until 2020-07-30
..

**Comment by Lanwalker** [comment]
I would like to have a history depository of older versions of their software, especially of their older clients for Windows and Android phones... I think the newer clients have stripped out some magnifying capabilities the older ones had... and their support sent me a junk copy with some real lousy chinese version included as a previous version.

**Reply by thirtythreeforty** [reply]
Re: old clients: shoot me an email; I have copies.

**Comment by twistedddx** [comment]
Lanwalker don't quote an entire post when it is not needed.


For versions of the Windows client there has been 4 main historical branches and 1 new current branch.
See below for a guide of versions, I have multiple examples of all.

You can still get 7.2.2.33 from here:
https://s3.amazonaws.com/reolink-storage/website/client/Reolink+Client+Windows+v7.2.2.33+x64.zip
You can get older versions of the Android app from the website apkmirror or similar.

Do you need a different version?
Reolink products started in 2017 and require a client post 2017.

Historical Windows versions:
a.b.c.d
a=distributor code (1=Generic, 2=Swann, 6=Amcrest, 7=Reolink)
b=major version
c=unknown but always 2
d=minor version

1)Major version 0. Known minor version range 22-35. era ~2013-2014:
Uses file and function names I would say looking very much like hikvision products.
Perhaps BC started their code trying to imitate hikvision SDK or perhaps their code started as a branch of hikvision code.
Does not support BC Modern protocol.

2)Major version 1. Known minor version range  1-28. era ~2014-2015
UI greatly overhauled
BC file naming. Internal functions still follow hikvision names but divergence starts.
Supports BC Modern protocol but does not support aac audio.

3)Major version 1. Known minor version 43-46. Era ~2016
This appears to be the start of major changes to the product.
These versions appear to be the start of the major version 2 works.
aac/mp4 support replacing adpcm/avi.
"Reolink" first appearance.

4)Major version 2. Known minor versions 7-33. Era ~2017-2020
Swann and all other distributors dropped.
Reolink only era with much of the code appearing to be substantially reviewed and patched.

Latest versions has a more typical version numbering:
a.b.c
a=major
b=minor
c=patch

5)Major version 8. Era 2020+
Complete rewrite of client

**Comment by thirtythreeforty** [comment]
@twistedddx Thanks for testing new versions. For things like this, should we start a wiki? I'd be happy to migrate documentation and known-good firmware versions over to it.

**Comment by richardklingler** [comment]
Hello

Just tested following cameras, which work out of the box:

- E1 Pro
- RLC-520, hardware model IPC_515B16M5M
- RLC-520, hardware model IPC_51516M5M

According to Reolink, the IPC_515B16M5M model is the new hardware version of the RLC-520...

Quote Reolink: "IPC_515B16M5M is our recently added hardware number."

Although RLC-520 and E1 Pro support RTSP I had more stable feeds when using the Reolink App on desktop or iOS than using RTSP.
